internetarchive/utils.py

```diff
@@ -398,3 +398,8 @@ def merge_dictionaries(dict0, dict1, keys_to_drop=None):
     # Items from `dict1` take precedence over items from `dict0`.
     new_dict.update(dict1)
     return new_dict
+
+
+def is_json(filename):
```

Comment:
Now that we have dropped legacy Python, let's add type hints on the function parameters and return type.
Also, given that `import os` is already done above, let's leverage that.
```
def is_json(file_path: str) -> bool:
    return os.path.splitext(file_path)[1] == ".json"
```
https://docs.python.org/3/library/os.path.html#os.path.splitext

Of course, my favorite would be `import pathlib` above and then:
```
def is_json(file_path: str) -> bool:
    return Path(file_path).suffix == ".json"
```
https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.suffix

Reply:
I'd love to make more use of pathlib now that we're PY3 only. With that in mind, it might be worth using it here vs. the simplified `file_path.endswith('.json')` for the sake of consistency. I'm all for type hints, sounds great.

Reply:
In my opinion the `os.path.splitext`solution is clearly inferior to `path.endswith`.

Yet if me managed to change to paths where possible and over time consistently manged to use pathlib, that would be pretty nice e.g. if functions where it made sense would already take a path as input instead of a string.